### PR TITLE
In preparation for Ecto 3.0

### PR DESCRIPTION
**Pull request by tarzan**
I wanted to try out Ecto 3.0 and needed to update phoenix_ecto too in order to make my app compile. 

These are the changes needed for compiling and passing the tests with the current master of Ecto. 

Hope this helps with future updates.

P.S. sorry about also applying the formatter, it is configured to automatically run on save in my editor.

### CODE REVIEW

**Comment by josevalim** [comment]
Thanks @tarzan! Unfortunately the formatter makes it really hard to know what changed. Is there a chance we can get a patch without the formatter? Thank you.

**Comment by tarzan** [comment]
I'll try to pull out the edits that matter. Github allows you to filter out the whitespace changes though so, https://github.com/phoenixframework/phoenix_ecto/pull/102/files?w=1 is a more compact diff.

File: lib/phoenix_ecto/html.ex [hunk 1]
```diff
@@ -278,8 +308,8 @@ if Code.ensure_loaded?(Phoenix.HTML) do
     end
   end
 
-  if Code.ensure_loaded?(Ecto.DateTime) do
-    defimpl Phoenix.HTML.Safe, for: [Ecto.Time, Ecto.Date, Ecto.DateTime] do
+  if Code.ensure_loaded?(DateTime) do
+    defimpl Phoenix.HTML.Safe, for: [Time, Date, DateTime] do
```
Comment (josevalim) [inline lib/phoenix_ecto/html.ex]: Those should rather be moved to phoenix_html. I will work on this.

Comment (josevalim) [reply lib/phoenix_ecto/html.ex]: It is already in phoenix_html. So we can just remove this as long as we bump the ecto requirement.

**Comment by tarzan** [comment]
- [x] re-apply changes without formatter
- [x] re-move HTML safe implementation for `Time`, `Date` and `DateTime`

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**